Testnet/Multisig/Multisig/MultisigContract.cs

```diff
@@ -0,0 +1,355 @@
+    private void SetOwners(byte[] addresses)
+    {
+        Address[] addressList = Serializer.ToArray<Address>(addresses);
+
+        foreach (var address in addressList)
+        {
+            SetOwner(address);
+        }
+
+        OwnerCount = (uint)addressList.Length;
+    }
+
+    private void SetOwner(Address address)
+    {
+        State.SetBool($"{OwnerPrefix}:{address}", true);
+
+        Log(new OwnerAddition() { Owner = address });
+    }
+
+    private void UnsetOwner(Address address)
+    {
+        State.SetBool($"{OwnerPrefix}:{address}", false);
+
+        Log(new OwnerRemoval() { Owner = address });
+    }
+
+    /// <summary>
+    /// Submits method and parameter metadata describing a contract call that the multisig contract should invoke once it is sufficiently confirmed.
+    /// </summary>
+    /// <param name="data">An array of method parameters encoded as packed byte arrays. See the <see cref="Transaction"/> struct for further information.</param>
+    /// <returns>The transactionId of the submitted multisig transaction.</returns>
+    /// <remarks>The submitter implicitly provides a confirmation for the submitted transaction.</remarks>
+    public ulong Submit(Address destination, string methodName, byte[] data)
+    {
+        EnsureOwnersOnly();
+
+        TransactionCount++;
+
+        State.SetStruct($"{TransactionPrefix}:{TransactionCount}", new Transaction()
+        {
+            Destination = destination,
+            Executed = false,
+            Value = 0,
+            MethodName = methodName,
+            Parameters = data
+        });
+
+        Log(new Submission() { TransactionId = TransactionCount });
+
+        Confirm(TransactionCount);
+
+        return TransactionCount;
+    }
+
+    /// <summary>
+    /// Called by contract owners to indicate that they accept the proposed method invocation provided via <see cref="Submit"/>.
+    /// </summary>
+    /// <param name="transactionId">The transactionId of the submitted multisig transaction.</param>
+    /// <remarks>Once the confirmation count matches or exceeds <see cref="Required"/> the contract call will be executed.</remarks>
+    public void Confirm(ulong transactionId)
+    {
+        Assert(IsOwner(Message.Sender), "Sender is not authorized to confirm.");
+        Assert(transactionId <= TransactionCount, "Transaction ID does not exist.");
+        Assert(!IsConfirmedBy(transactionId, Message.Sender), "Sender has already confirmed the transaction.");
+
+        State.SetBool($"{ConfirmationPrefix}:{transactionId}:{Message.Sender}", true);
+
+        uint confirmationCount = State.GetUInt32($"{ConfirmationCountPrefix}:{transactionId}");
+
+        State.SetUInt32($"{ConfirmationCountPrefix}:{transactionId}", ++confirmationCount);
+
+        Log(new Confirmation() { Sender = Message.Sender, TransactionId = transactionId });
+
+        if (confirmationCount < Required)
+            return;
+
```

Comment:
Consider moving the code below here into its own `Execute` method. Partly for readability, though maybe would be possible to allow re-execution of failed transactions.